### PR TITLE
Add unit tests for workflow scheduler

### PR DESCRIPTION
**Pull request by nikhitashanker77**
-Add 
1. test for parseEventXML in utils
2. test for operations until adding context to the guid to context map for a subscription
3. test for running the workflow on a callback
4. end to end test that needs to be run manually

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	// Action
+	var mainObj *Main = newMain()
+	mainObj.setupWorkflows(schedulesJson, workflowFactory)
+
+	// Assert
+	// Assert that the GUID to context map has only 1 context.
+	var workflowScheduler *WorkflowScheduler = mainObj.WorkflowScheduler
+	assert.Equal(t, len(workflowScheduler.guidToContext), 1)
+	
+	// Check that the first and second nodes in the context's workflow are of type DummyNodeA by checking the value of result field in the objects.
```
Comment (MichaelBehm) [inline main_test.go]: Move this to the workflow_scheduler_tester.go

Comment (MichaelBehm) [reply main_test.go]: As a test for the `setupWinEventSchedule` function

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		log.Fatal(err)
+	}
+
+	// Action
+	var mainObj *Main = newMain()
+	mainObj.setupWorkflows(schedulesJson, workflowFactory)
+
+	// Assert
+	// Assert that the GUID to context map has only 1 context.
+	var workflowScheduler *WorkflowScheduler = mainObj.WorkflowScheduler
```
Comment (MichaelBehm) [inline main_test.go]: Mock the addWinEventSchedule function, just check if we call it X times here

Comment (MichaelBehm) [reply main_test.go]: (this is to avoid testing WorkflowScheduler)

File: README.md [hunk 3]
```diff
@@ -41,6 +41,10 @@ To run all tests use:
 cd BladeMonRT
 go test ./...
```
Comment (MichaelBehm) [inline README.md]: See if we can exclude E2E test, if so, let's make it so people don't have to change the filename

File: README.md [hunk 4]
```diff
@@ -41,6 +41,10 @@ To run all tests use:
...
 
+To run the end-to-end test:
+rename test_end_to_end.go to test_end_to_end_test.go
+go test -run TestEndToEnd
```
Comment (MichaelBehm) [inline README.md:45]: Explain what this does:
- Runs BRT until a keyboard interrupt
- you will have to manually raise ETW events in a separate Terminal

File: utils/utils.go [hunk 5]
```diff
@@ -10,6 +10,18 @@ import (
   "fmt"
 )
 
+/** Class that represents an event parsed from event XML. */
+type EventFromXML struct {
```
Comment (MichaelBehm) [inline utils/utils.go]: Double check this

**Review by MichaelBehm — CHANGES_REQUESTED** [review]
See my Comments

**Comment by nikhitashanker77** [comment]
https://github.com/microsoft/BladeMonRT/pull/10#discussion_r688095426

Used short flag to stop the end-to-end test from being run with all other tests. No need to change filename now.

File: README.md [hunk 6]
```diff
@@ -37,9 +37,16 @@ To run the main use:
...
 go run .
 
-To run all tests use:
+To run all tests except the end-to-end test use:
 cd BladeMonRT
-go test ./...
+go test ./... -short
+
+To run the end-to-end test:
+rename test_end_to_end.go to test_end_to_end_test.go
```
Comment (MichaelBehm) [inline README.md]: nit: remove this line

Comment (MichaelBehm) [reply README.md]: No longer necessary to rename the end to end test file, right?

**Review by MichaelBehm — APPROVED** [review]
Looks good! See my 1 nit on README.md

**Comment by nikhitashanker77** [comment]
https://github.com/microsoft/BladeMonRT/pull/10#discussion_r688784982 Yes no longer necessary to rename the file to run it. I have removed this from the README.md

**Pull request merged**